niworkflows/interfaces/bids.py

```diff
@@ -616,7 +617,10 @@ def _run_interface(self, runtime):
 
                 if data_dtype == "source":  # match source dtype
                     try:
-                        data_dtype = nb.load(self.inputs.source_file).get_data_dtype()
+                        source_file = self.inputs.source_file
+                        if isinstance(source_file, Sequence) and not isinstance(source_file, str):
+                            source_file = source_file[0]
+                        data_dtype = nb.load(source_file).get_data_dtype()
```

Comment:
`self.inputs.source_file` should always be a `list`.

```suggestion
                        data_dtype = self.inputs.source_file[0].get_data_dtype()
```